src/CachedRemoteRequest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Class CachedRemoteRequest.
+ *
+ * @package Amp\AmpWP
+ */
+
+namespace Amp\AmpWP;
+
+use Amp\Exception\FailedToFetchFromRemoteUrl;
+use Amp\RemoteRequest;
+
+/**
+ * Caching decorator for RemoteRequest implementations.
+ *
+ * Caching uses WordPress transients.
+ *
+ * @package Amp\AmpWP
+ */
+final class CachedRemoteRequest implements RemoteRequest {
+
+
+	/**
+	 * Remote request object to decorate with caching.
+	 *
+	 * @var RemoteRequest
+	 */
+	private $remote_request;
+
+	/**
+	 * Cache expiration time in hours.
+	 *
+	 * @var int
+	 */
+	private $expiry;
+
+	/**
+	 * Instantiate a CachedRemoteRequest object.
+	 *
+	 * This is a decorator that can wrap around an existing remote request object to add a caching layer.
+	 *
+	 * @param RemoteRequest $remote_request Remote request object to decorate with caching.
+	 * @param int           $expiry         Optional. Cache expiry in hours. Defaults to 24 hours.
+	 */
+	public function __construct( RemoteRequest $remote_request, $expiry = 24 ) {
```

Comment:
Why is `$expiry` not expressed in seconds?

Reply:
Hours sounded granular enough, but I can change to seconds if you prefer.

Reply:
It's just that seconds are more commonly used for this, I think. For example, transients use seconds for expiration. Unix timestamps are in seconds. By using seconds, it will be more flexible for other use cases perhaps.